File: modules/secure-recorder/ios/State.swift
import Foundation

/**
 * Immutable recording state (thread-safe)
 */
internal struct State {
  internal let isRecording: Bool
  internal let sessionId: String?
  internal let filePath: String?
  
  internal static let idle = State(isRecording: false, sessionId: nil, filePath: nil)
}

File: modules/secure-recorder/android/src/main/java/expo/modules/securerecorder/State.kt
package expo.modules.securerecorder

/**
 * Immutable recording state (thread-safe)
 */
data class State(
  internal val isRecording: Boolean,
  internal val sessionId: String?,
  internal val filePath: String?
) {
  companion object {
    internal val IDLE = State(false, null, null)
  }
}
